File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.rb
# @param {String} s
# @return {Integer}
def second_highest(s)
    res=[]
    s.chars.each do |c|
        p c
        res << c if c.match?(/\d/)
    end
    if res.uniq.size > 1
        res.delete(res.max) 
        res.max.to_i
    else
        return -1
    end
end

File: 1796-second-largest-digit-in-a-string/1796-second-largest-digit-in-a-string.py
class Solution:
    def secondHighest(self, s: str) -> int:
        res=[]
        for c in s:
            if c.isdigit():
                res.append(c)   

        if len(set(res)) > 1:
            res.remove(max(res))
            return int(max(res))
        else:
            return -1     



